File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        s = list(s)
        t = list(t)
        s.sort()
        t.sort()
        
        for i in range(len(s)):
            if s[i] != t[i]:
                return False
        return True
        

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> hashmap1 = new HashMap<>();
        Map<Character, Integer> hashmap2 = new HashMap<>();


        for (char c : s.toCharArray()) {
            if (hashmap1.containsKey(c)) {
                hashmap1.put(c, hashmap1.get(c) + 1);
            }
            else {
                hashmap1.put(c, 1);
            }
        }

        for (char c : t.toCharArray()) {
            if (hashmap2.containsKey(c)) {
                hashmap2.put(c, hashmap2.get(c) + 1);
            }
            else {
                hashmap2.put(c, 1);
            }
        }
        
        if (hashmap2.equals(hashmap1)) {
            return true;
        }
        return false; 
    }
}